Lib/csv.py

```diff
@@ -415,25 +417,38 @@ def has_header(self, sample):
             if len(row) != columns:
                 continue # skip rows that have irregular number of columns
 
+            # checking if all col are strings
+            for cols in list(columnTypes.keys()):
```

Comment:
dicts are directly iterable!  and a list is not necessary here.

```suggestion
            for cols in columnTypes:
```